### PR TITLE
ci: daily release artifacts for the snapshots

### PR DESCRIPTION
**Pull request by v1v**
## What is the problem this PR solves?

Enable the Daily Release Artifacts for the Beats project

## How does this PR solve the problem?

A specific pipeline in charge of publishing the snapshots on a merge-commit only

## Further details

This new stage will be triggered automatically after a build in the CI for the active branches has finished successfully.

1) Only once the e2e-testing has successfully finished then (this is our quality gate)
2) A following stage that combines them and runs a release-manager container to upload them to GCS.
3) Notification in Slack, and email, if (and only if) the pipeline fails.

## Additional changes

* `get-version` Makefile goal to avoid requiring `mage`, in fact, the APM Server uses a similar approach -> https://github.com/elastic/apm-server/blob/f1be74a473090bee3208722f3578616850a8c7c5/Makefile#L29 while the Fleet-Server uses something similar but different -> https://github.com/elastic/fleet-server/blob/main/Makefile#L3
* Enable sha512 file for the dockerlogbeat component
* Create dependencies and dashboards in the parallel package stage.

## Test

image

This [build](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackaging/detail/PR-30897/56/pipeline) 
produced https://artifacts-snapshot.elastic.co/beats/8.2.0-391ea681/summary-8.2.0-SNAPSHOT.html

### CODE REVIEW

File: .ci/scripts/release-manager.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+docker run --rm \
+  --name beats \
+  -e VAULT_ADDR \
+  -e VAULT_ROLE_ID \
+  -e VAULT_SECRET_ID \
+  --mount type=bind,readonly=false,src="$PWD",target=/artifacts \
+  "$IMAGE" \
+    cli collect \
+      --project beats \
+      --branch "main" \
```
Comment (v1v) [inline .ci/scripts/release-manager.sh:47]: TBC

```suggestion
      --branch "$BRANCH_NAME" \
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -82,11 +87,10 @@ pipeline {
...
             // Stash without any build/dependencies context to support different architectures.
             stashV2(name: 'source', bucket: "${JOB_GCS_BUCKET_STASH}", credentialsId: "${JOB_GCS_CREDENTIALS}")
-            withMageEnv(){
-              dir("${BASE_DIR}"){
-                setEnvVar('BEAT_VERSION', sh(label: 'Get beat version', script: 'make get-version', returnStdout: true)?.trim())
-              }
+            dir("${BASE_DIR}"){
+              setEnvVar('BEAT_VERSION', sh(label: 'Get beat version', script: 'make get-version', returnStdout: true)?.trim())
             }
+            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable('main'))
```
Comment (v1v) [inline .ci/packaging.groovy:93]:
```suggestion
            setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
```

File: .ci/packaging.groovy [hunk 3]
```diff
@@ -23,9 +23,11 @@ pipeline {
     GITHUB_CHECK_E2E_TESTS_NAME = 'E2E Tests'
     SNAPSHOT = "true"
     PIPELINE_LOG_LEVEL = "INFO"
+    SLACK_CHANNEL = '#observablt-bots'
+    NOTIFY_TO = 'package-beats+victor.martinez@elastic.co'
```
Comment (v1v) [inline .ci/packaging.groovy:27]:
```suggestion
    SLACK_CHANNEL = '#beats'
    NOTIFY_TO = 'package-beats+beats-contrib@elastic.co'
```

File: .ci/scripts/release-manager.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      --commit "$(git rev-parse HEAD)" \
+      --workflow "snapshot" \
+      --artifact-set main \
+      --version "${VERSION}" | tee release-manager-report.out
```
Comment (v1v) [inline .ci/scripts/release-manager.sh:62]:
```suggestion
      --version "${VERSION}"
```

**Review by mdelapenya — APPROVED** [review]

File: .ci/packaging.groovy [hunk 5]
```diff
@@ -23,9 +23,11 @@ pipeline {
     GITHUB_CHECK_E2E_TESTS_NAME = 'E2E Tests'
     SNAPSHOT = "true"
     PIPELINE_LOG_LEVEL = "INFO"
+    SLACK_CHANNEL = '#beats'
+    NOTIFY_TO = 'package-beats+beats-contrib@elastic.co'
```
Comment (v1v) [inline .ci/packaging.groovy:27]:
```suggestion
    NOTIFY_TO = 'beats-contrib+package-beats@elastic.co'
```

**Review by cmacknz — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**